src/restLink.ts

```diff
@@ -84,28 +85,40 @@ export const validateRequestMethodForOperationType = (
   }
 };
 
+let exportVariables = {};
+
 const resolver = async (fieldName, root, args, context, info) => {
   const { directives, isLeaf, resultKey } = info;
+  if (root === null) {
+    exportVariables = {};
+  }
   if (isLeaf) {
-    return root[resultKey];
+    const leafValue = root[resultKey];
+    if (directives && directives.export) {
+      exportVariables[directives.export.as] = leafValue;
+    }
+    return leafValue;
   }
-  const { credentials, endpoints, headers } = context;
+  const { credentials, endpoints, headers, customFetch } = context;
   const { path, endpoint } = directives.rest;
   const uri = getURIFromEndpoints(endpoints, endpoint);
   try {
-    let pathWithParams = path;
-    if (args) {
-      pathWithParams = Object.keys(args).reduce(
-        (acc, e) => replaceParam(acc, e, args[e]),
-        path,
+    const argsWithExport = { ...args, ...exportVariables };
+    let pathWithParams = Object.keys(argsWithExport).reduce(
+      (acc, e) => replaceParam(acc, e, argsWithExport[e]),
+      path,
+    );
+    if (pathWithParams.includes(':')) {
+      throw new Error(
+        'Missing params to run query, specify it in the query params or use an export directive',
```

Comment:
Can we log the Params that are missing instead of having a generic message?

Reply:
Also, is there some way to indicate that this error applies to a specific nested query? I worry about complex calls with many levels of queries and users not knowing which one faulted.

Reply:
Good, I will add the param name, for the nested query, yes I think we can and it is not that difficult to do